src/openxr/OpenXRRuntimeSelect.cpp

```diff
@@ -0,0 +1,93 @@
+/////////////////////////////////////////////////////////////////////////////////////
+// Our OpenXR runtime select GDNative object, this allows us to select the runtime
+
+#include "OpenXRRuntimeSelect.h"
+
+using namespace godot;
+
+void OpenXRRuntimeSelect::_register_methods() {
+	register_method("set_runtime", &OpenXRRuntimeSelect::set_runtime);
+	register_method("get_runtime", &OpenXRRuntimeSelect::get_runtime);
+}
+
+OpenXRRuntimeSelect::OpenXRRuntimeSelect() {
+	// nothing to do here
+}
+
+OpenXRRuntimeSelect::~OpenXRRuntimeSelect() {
+	// nothing to do here
+}
+
+void OpenXRRuntimeSelect::_init() {
+	// nothing to do here
+}
+
+#ifdef WIN32
+String OpenXRRuntimeSelect::format_registry_error(DWORD p_err) const {
+	char error[1024];
+
+	DWORD result = FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM, NULL, p_err, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), error, 1024, NULL);
+	if (result != 0) {
+		return String(error);
+	} else {
+		return String("Unknown - ") + String::num_int64(p_err);
+	}
+}
+#endif
+
+String OpenXRRuntimeSelect::get_runtime() const {
+#ifdef WIN32
+	String return_value("n/a");
+
+	// TODO possibly rewrite to use RegOpenKeyExW/RegQueryValueExW
+	HKEY key;
+	DWORD result = RegOpenKeyExA(HKEY_LOCAL_MACHINE, "SOFTWARE\\Khronos\\OpenXR\\1", 0, KEY_READ, &key);
+	if (result == ERROR_SUCCESS) {
+		char path[1024];
+		DWORD len = 1024;
+		result = RegQueryValueExA(key, "ActiveRuntime", NULL, NULL, (BYTE *)path, &len);
+		if (result == ERROR_SUCCESS) {
+			return_value = path;
+		} else {
+			Godot::print("OpenXRRuntimeSelect.get_runtime(): Can't read registry key: {0}", format_registry_error(result));
+		}
+
+		RegCloseKey(key);
+	} else {
+		Godot::print("OpenXRRuntimeSelect.get_runtime(): Can't open registry key: {0}", format_registry_error(result));
+	}
+
+	return return_value;
+#else
+	Godot::print("OpenXRRuntimeSelect.get_runtime(): Unsupported platform");
+	return String("n/a");
+#endif
+}
+
+bool OpenXRRuntimeSelect::set_runtime(const String p_runtime) {
+#ifdef WIN32
+	// TODO possibly rewrite to use RegOpenKeyExW/RegSetValueExW
```

Comment:
Should have removed this comment :P 

I'm leaving this function here even though it will fail due to rights issues. Possibly in the future we should investigate if we can elevate the plugin at least in the editor.